apps/src/pythonlab/pyodideRunner.ts

```diff
@@ -43,6 +43,7 @@ export async function handleRunClick(
     }
     consoleManager?.writeSystemMessage('Running program...', appName);
     await runPythonCode(code, source);
+    CodebridgeRegistry.getInstance().getNeighborhood()?.onClose();
```

Comment:
Here and below, `neighborhood` calls only executed if `neighborhood` exists, i.e., if the level's `miniApp` is `neighborhood`. 
https://github.com/code-dot-org/code-dot-org/blob/071a7e22dfedbc32c3b1af0a217beac1841847f9/apps/src/codebridge/MiniAppPreview/NeighborhoodPreview.tsx#L34-L46

Reply:
Do we need to clean up the instance when we switch levels (between a neighborhood and non-neighborhood level)?

Reply:
Oh interesting. I didn't realize that `levelProperties` as a dependency wouldn't trigger this `useEffect` when switching levels. I can add `currentLevelId` as a dependency.
Could you explain why `levelProperties` does not update on a level change?

Reply:
hmm level properties should update on level change

Reply:
We never actually remove the neighborhood instance from the registry though, so it could stick around when switching levels, independent of whether the useEffect triggers, right?

Reply:
discussed offline, will handle separately

Reply:
Removed `currentLevelId` dependency.
Updated code so that `neighborhood`-related code run only when `miniApp` is `neighborhood`. 
Screencast of user switching from non-neighborhood -> neighborhood -> non-neighborhood level. Can run code successfully.

https://github.com/user-attachments/assets/ff4f5eb1-2884-4db4-bbcd-0e3f9f41f252

Reply:
Thanks so much for flagging this!